[FEATURE] Aligner le style du PixTooltip avec le design system (PIX-3019).

AndreiaPena opened the pull request:
## :unicorn: Problème
Les composants Pix UI ne correspondent pas tous à ce qui est défini dans le design system. Il faut également veiller à ce qu'ils soit accessibles et responsive.

Le composant PixTooltip ne correspond pas avec le design system sur les points suivants :

- Couleur de fond pour le mode foncé.

![image](https://user-images.githubusercontent.com/516360/131646232-3fbd5d68-49fe-4336-856c-697defc5f35a.png)

- Bordure pour le mode clair.

![image](https://user-images.githubusercontent.com/516360/131646263-08ead7aa-c14d-40ed-9cf4-42be995f9c4c.png)

## :rainbow: Remarques
- Le tooltip est désormais visible au focus.
- Cacher la tooltip dans le DOM (`display: none`) pour éviter d'avoir le focus sur des éléments à l'intérieur de la tooltip quand celle-ci n'est pas affichée.

Des éléments d'accessibilité ont été ajouté au niveau du composant ont été ajouté (ex: `role="tooltip"`).

Également une documentation plus précise des éléments d'a11y à prendre en compte à l'utilisation de `PixTooltip` : 

![image](https://user-images.githubusercontent.com/516360/131646731-c333e0b4-f045-421c-a5e9-8a94177c8c16.png)


## :100: Pour tester
Lien vers Zeroheight https://zeroheight.com/8dd127da7/p/27df2f-tooltips et Storybook https://ui-pr129.review.pix.fr

AndreiaPena committed:
Update PixTooltip style

pix-service commented:
I'm deploying this PR to:

- Storybook: https://ui-pr129.review.pix.fr

 Please check it out!

AndreiaPena committed:
Update PixTooltip documentation

bpetetot reviewed (approved):
👍

bpetetot committed:
Update PixTooltip light style

bpetetot committed:
Hide tooltip in the DOM

To avoid its content (like links, button) to be focusable with keyboard navigation

bpetetot committed:
Add A11y on PixTooltip with some documentation

Anne-Gaelle-S commented on addon/stories/pix-tooltip.stories.js:
```diff
@@ -10,13 +10,62 @@ export const tooltip = (args) => {
+Default.args = {
+  text: 'Hello World',
+  label: 'À survoler pour voir la tooltip',
+};
```
**question**
`label` n'est pas un argument de la PixTooltip, pourquoi l'avoir surchargé dans les `args` plutôt que l'avoir mis "en dur" entre les balises de PixButton ?

AndreiaPena replied:
J'ai fait ce choix pour pouvoir renommer le contenu des buttons dans la doc :)

xav-car commented:
**global/question :** Ne devrait-on pas arrêter les balises div en main pour nos composants dans les cas ou l'on doit l'insérer dans un `Phrasing content` . le div n'est pas sémantiquement autorisé.

(désolé si ça a été évoqué dans un wiki ou autre)

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories

bpetetot commented:
> **global/question :** Ne devrait-on pas arrêter les balises div en main pour nos composants dans les cas ou l'on doit l'insérer dans un `Phrasing content` . le div n'est pas sémantiquement autorisé.
> 
> (désolé si ça a été évoqué dans un wiki ou autre)
> 
> https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories

En effet, et dans le cas de la tooltip le `span` semble plus judicieux, par contre on est obligé de conserver un élément main pour gérer le style du composant et son comportement.

bpetetot committed:
Change PixTooltip root element to span
